sdk/examples/devmode_rust/src/main.rs

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2018 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ------------------------------------------------------------------------------
+ */
+
+#[macro_use]
+extern crate clap;
+#[macro_use]
+extern crate log;
+extern crate log4rs;
+extern crate sawtooth_sdk;
+
+use std::sync::mpsc::{RecvTimeoutError, Receiver};
+use std::time;
+use std::process;
+
+use log::LogLevelFilter;
+use log4rs::append::console::ConsoleAppender;
+use log4rs::config::{Appender, Config, Root};
+use log4rs::encode::pattern::PatternEncoder;
+
+use sawtooth_sdk::consensus::{
+    service::Service,
+    engine::*,
+    driver::Driver,
+    zmq_driver::ZmqDriver,
+};
+
+pub struct Devmode {
+    period: time::Duration,
+    exit: Exit,
+}
+
+impl Devmode {
+    fn new(period: time::Duration) -> Self {
+        Devmode {
+            period: period,
+            exit: Exit::new(),
+        }
+    }
+}
+
+impl Engine for Devmode {
```

Comment:
If we're following the transaction handler model, the engine and `main` should be in separate files.